packages/preview-middleware-client/src/utils/version.ts

```diff
@@ -70,6 +70,6 @@ export function isLowerThanMinimalUi5Version(
  * @param ui5VersionInfo to be mentioned in the message
  * @returns string with validation message.
  */
-export function getUI5VersionValidationMessage(ui5VersionInfo: Ui5VersionInfo): string {
+export function getUI5VersionValidationMessage(ui5VersionInfo: Ui5VersionInfo) {
```

Comment:
Why the return type was removed from here?

Reply:
Because TypeScript infers the return type anyway. I thought this is a general TS feature and not IDE specific. At least in my IDE `:string` is superfluous. If that assumption proves wrong I can of course add it again

Reply:
We don't have a clear guideline for that in this repo (I think). Having the type there has the advantage that when you change the code and accidentally forgot to return something in an branch then your IDE will show an error directly here, otherwise, it would show errors where it is consumed e.g. if an undefined check is missing.

I prefer having return value explicitly typed because it helps people that don't know the code that well but, again, it is not a rule in this repo (I think).

Reply:
The reason for bringing this up is that there is an [ESLint rule](https://typescript-eslint.io/rules/explicit-function-return-type/) for this and somewhat recently in one of the repositories complaining about this. It looks like it wasn't this one or it is no longer the case here, because I don't see the warnings anymore.

Reply:
Ok, then I'll add it again 👍🏻

Reply:
#2252 to have guidance in the future